### ISSUE TITLE
Local Logistics UI shows incorrect resources available for transfer

### ISSUE THREAD

**Issue by fat-lobyte** [BobPalmer/MKS#1100]
tl;dr: An indexing typo (probably) has the transferable resources all garbled up

When cycling through vessels, the list of resources available for transfer is wrong most of the time.

Assume you have three vessels.
1: MetallicOre, Substrate
2: Substrate, Uranitite
3: Uranitite, MetallicOre

Then you would have the following transferrable resources for each pair of vessels:
1 <-> 2: Substrate
1 <-> 3: MetallicOre
2 <-> 3: Uranitite

This is not what is displayed however. To reproduce, load the following save file and observe:

https://dl.dropboxusercontent.com/u/1414175/ksp/local-logistics-bug.sfs

Load the save, open up local logistics, and observe the following:

If you leave the right vessel at (3):
1 <-> 3: Uranitite. But Vessel (1) doesn't have that, so it's shown as 0.0/0.0
2 <-> 3: MetallicOre AND Uranitite. Uranitite is indeed shared between the vessels, but not MetallicOre which is 0.0/0.0 on (2).

If you cycle backwards now (from 2 <-> 3 to 1 <-> 3), what we get now is MetallicOre, which is correct for 1 <-> 3.

I think the display for the transferable resources lags behind by 1 when cycling vessels, except after the index of the right vessel (for example right selector is set to 2, then left selector is cycled to 3). Then, the list of resources of the vessels is compared to the same vessel, which yields all resources in the right vessel (for example 2), which may or may not exist in the left vessel.

It seems there is some indexing mayhem in the code, but I have a feeling that it should be rather easy to fix.
